presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -404,7 +405,11 @@ public PlanOptimizers(
                 stats,
                 statsCalculator,
                 costCalculator,
-                ImmutableSet.of(new RemoveRedundantIdentityProjections())));
+                ImmutableSet.of(
+                        new RemoveRedundantIdentityProjections(),
+                        new TransformSpatialPredicateToJoin(metadata),
+                        new InlineProjections(),
+                        new RemoveRedundantIdentityProjections())));
```

Comment:
Ironically, one of the instances of RemoveRedundantIdentityProjections is redundant :) Remove it.